cmd/tink-worker/client/client_test.go

```diff
@@ -0,0 +1,222 @@
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return &workflow.GetWorkflowDataResponse{}, nil
+}
+
+func (s *fakeServer) GetWorkflowDataVersion(context.Context, *workflow.GetWorkflowDataRequest) (*workflow.GetWorkflowDataResponse, error) {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return &workflow.GetWorkflowDataResponse{}, nil
+}
+
+func (s *fakeServer) GetWorkflowContext(context.Context, *workflow.GetRequest) (*workflow.WorkflowContext, error) {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return &workflow.WorkflowContext{}, nil
+}
+
+func (s *fakeServer) GetWorkflowContexts(*workflow.WorkflowContextRequest, workflow.WorkflowService_GetWorkflowContextsServer) error {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return nil
+}
+
+func (s *fakeServer) GetWorkflowContextList(context.Context, *workflow.WorkflowContextRequest) (*workflow.WorkflowContextList, error) {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return &workflow.WorkflowContextList{}, nil
+}
+
+func (s *fakeServer) ListWorkflows(*workflow.Empty, workflow.WorkflowService_ListWorkflowsServer) error {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return nil
+}
+
+func (s *fakeServer) ShowWorkflowEvents(*workflow.GetRequest, workflow.WorkflowService_ShowWorkflowEventsServer) error {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return nil
+}
+
+func (s *fakeServer) UpdateWorkflowData(context.Context, *workflow.UpdateWorkflowDataRequest) (*workflow.Empty, error) {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return &workflow.Empty{}, nil
+}
+
+func (s *fakeServer) ReportActionStatus(context.Context, *workflow.WorkflowActionStatus) (*workflow.Empty, error) {
+	// Dummy response, we only need this definition to satisfy the interface requirements
+	return &workflow.Empty{}, nil
+}
+
+const bufSize = 1024 * 1024
+
+func startWorkflowServerAndConnectClient(t *testing.T, name string, server *grpc.Server) workflow.WorkflowServiceClient {
+	t.Helper()
+
+	ctx := context.Background()
+	listener := bufconn.Listen(bufSize)
+	go func() {
+		t.Helper()
+
+		if err := server.Server().Serve(listener); err != nil {
+			t.Error(fmt.Errorf("%s.Serve exited with error: %w", name, err))
+		}
+	}()
+
+	dialer := func(ctx context.Context, _ string) (net.Conn, error) {
+		return listener.DialContext(ctx)
+	}
+
+	conn, err := ggrpc.DialContext(ctx, "bufnet", ggrpc.WithContextDialer(dialer), ggrpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Failed to dial %s: %v", name, err)
+	}
+
+	client := workflow.NewWorkflowServiceClient(conn)
+	return client
+}
+
+func TestObtainTinkServerCreds(t *testing.T) {
```

Comment:
Food for thought, this function really feels like it could benefit from the use of table tests. https://github.com/golang/go/wiki/TableDrivenTests

Reply:
Cool, I gave that a try and added table tests here.